### PR TITLE
[1.19] Automated cherry pick of fixes for "large leases overload event etcd" issue (96836)

### PR DESCRIPTION
**Pull request by mborsz**
Cherry pick of #97009 #97480 #98257 on release-1.19.

#97009: apiserver add --lease-reuse-duration-seconds to config lease
#97480: apiserver add metric etcd_lease_object_counts
#98257: api-server add --lease-max-object-count

All those three PRs are a fix for an issue where creating a large number of events was causing event-etcd overload (https://github.com/kubernetes/kubernetes/issues/96836).

```release-note
Fixed a bug where a high churn of events was causing master instability by reducing the maximum number of objects (events) attached to a single etcd lease.
```

### CODE REVIEW

**Comment by mborsz** [comment]
https://github.com/kubernetes/kubernetes/pull/100084 for 1.20 cherry pick

File: staging/src/k8s.io/apiserver/pkg/server/options/etcd.go [hunk 1]
```diff
@@ -180,8 +180,10 @@ func (s *EtcdOptions) AddFlags(fs *pflag.FlagSet) {
 
 	fs.DurationVar(&s.StorageConfig.DBMetricPollInterval, "etcd-db-metric-poll-interval", s.StorageConfig.DBMetricPollInterval,
 		"The interval of requests to poll etcd and update metric. 0 disables the metric collection")
-}
 
+	fs.Int64Var(&s.StorageConfig.LeaseManagerConfig.ReuseDurationSeconds, "lease-reuse-duration-seconds", s.StorageConfig.LeaseManagerConfig.ReuseDurationSeconds,
+		"The time in seconds that each lease is reused. A lower value could avoid large number of objects reusing the same lease. Notice that a too small value may cause performance problems at storage layer.")
+}
```
Comment (wojtek-t) [inline staging/src/k8s.io/apiserver/pkg/server/options/etcd.go:186]: nit: empty line after } and no empty line before this flag

**Review by wojtek-t — COMMENTED** [review]
One minor nit

File: staging/src/k8s.io/apiserver/pkg/server/options/etcd.go (hunk 1 above)
Comment (mborsz) [reply staging/src/k8s.io/apiserver/pkg/server/options/etcd.go:186]: Added empty line after }

Please note that there is only one line before this flag (line. 183), which matches other flags in this file.

The "new line" in 184 is simply github's line wrapping of 184 line into two lines (you can use "..." > "View file" to see the full file).

**Comment by wojtek-t** [comment]
/lgtm
/approve

@kubernetes/release-managers - for cherrypick approval

**Pull request merged**